ethcore/src/snapshot/service.rs

```diff
@@ -412,7 +424,11 @@ impl Service {
 					next_chain.insert_unordered_block(&mut batch, block, block_receipts, Some(parent_total_difficulty), false, true);
 					count += 1;
 				},
-				_ => break,
+				_ => {
+					error!(target: "snapshot", "migrate_blocks: failed to find receipts and parent total difficulty. Block #{}, parent_hash={:?}, parent_total_difficulty={:?}",
+						block_number, parent_hash, parent_total_difficulty);
+					break
```

Comment:
should we return an error here instead of breaking out of the loop?

Reply:
The error is returned further down (when `target_hash != parent_hash`) but honestly I don't know why we want to flush to the DB right before returning an error. Maybe @ngotchac knows?

Reply:
Hm no reason really, I think initially it might have been so you don't have to return the same error twice, but I guess it's the only case where it would error bellow so you could do it right there.